package/MDAnalysis/core/groups.py

```diff
@@ -1876,6 +1954,67 @@ def __getattr__(self, attr):
         raise AttributeError("{cls} has no attribute {attr}"
                              "".format(cls=self.__class__.__name__, attr=attr))
 
+class UpdatingAtomGroup(AtomGroup):
+    """:class:`AtomGroup` subclass that dynamically updates its selected atoms.
+
+    Accessing any attribute/method of an :class:`UpdatingAtomGroup` instance
+    triggers a check for the last frame the group was updated. If the last
+    frame matches the current trajectory frame, the attribute is returned
+    normally; otherwise the group is updated (the stored selections are
+    re-applied), and only then is the attribute returned.
+
+    .. versionadded:: 0.16.0
+
+    """
+
+    def __new__(cls, *args, **kwargs):
+        # If we're reinstantiated in a non-updating way, we must become a
+        # regular AtomGroup. We check to see if we have an AtomGroup passed as
+        # either the 'base_group' kwarg, or the first arg, in which case we
+        # become an UpdatingAtomGroup.
+        bg = kwargs.get('base_group', args[0] if args else None)
+        if isinstance(bg, AtomGroup):
+            return super(UpdatingAtomGroup, cls).__new__(cls, *args, **kwargs)
+        return AtomGroup(*args, **kwargs)
+
+    def __init__(self, base_group, selections):
+        """
+
+        Parameters
+        ----------
+        base_group : :class:`AtomGroup`
+            group on which *selections* are to be applied.
+        selections : a tuple of :class:`~MDAnalysis.core.selection.Selection` instances
+            selections ready to be applied to *base_group*.
+
+        """
+        # Because we're implementing __getattribute__, which needs _u for
+        # its check, no self.attribute access can be made before this line
+        self._u = base_group._u
+        self._base_group = base_group
```

Comment:
Thinking forwards about pickling, if an `UpdatingGroup` just pickled the base group with itself, then even if `base_group` were also dynamic it would just recursively pickle until it hit a real `AtomGroup`.